app/src/main/java/com/github/polybooks/SaleInformationActivity.kt

```diff
@@ -0,0 +1,45 @@
+package com.github.polybooks
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+import com.github.polybooks.core.Sale
+import com.github.polybooks.utils.StringsManip
+
+class SaleInformationActivity: AppCompatActivity() {
+    companion object {
+        const val EXTRA_SALE_INFORMATION: String = "EXTRA_SALE_INFORMATION"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_sale_information)
+
+        val viewTitle: TextView = findViewById(R.id.sale_information_title)
+        val viewEdition: TextView = findViewById(R.id.sale_information_edition)
+        val viewAuthors: TextView = findViewById(R.id.sale_information_authors)
+        // val countryFlag: TextView = findViewById(R.id.countryFlag)
+        val viewPublishDate: TextView = findViewById(R.id.sale_information_book_publish_date)
+        val viewPublisher: TextView = findViewById(R.id.sale_information_book_publisher)
+        val viewFormat: TextView = findViewById(R.id.sale_information_book_format)
+        // val bookImage = findViewById(R.id.proof_picture)
+
+        val viewCondition: TextView = findViewById(R.id.sale_information_condition)
+        val viewPrice: TextView = findViewById(R.id.sale_information_price)
+
+
+        val sale = (intent.getSerializableExtra(EXTRA_SALE_INFORMATION) as Sale)
+
+        viewTitle.text = sale.book.title
+        viewEdition.text = sale.book.edition
+        viewAuthors.text = StringsManip.listAuthorsToString(sale.book.authors)
+
+        viewPublishDate.text = sale.book.publishDate.toString()
+
+        viewPublisher.text = sale.book.publisher
+        viewFormat.text = sale.book.format
+
+        viewCondition.text = sale.condition.name
+        viewPrice.text = sale.price.toString()
```

Comment:
I would suggest to put all those in something like `setFieldsTexts` function so onCreate is not too long. 
You can further more set all the TextViews in another function as well, and call the two functions in onCreate

Reply:
I unfortunately don't think it is possible :/ 
Creating such function would have to take both the `TextView` and its value in parameter (in total 16 parameters ^^'

Am I missing something? Coz' I would also love to make it "more clean" ^^